benches/benches/bevy_ecs/components/change_detection.rs

```diff
@@ -85,17 +100,17 @@ fn all_changed_detection_generic<T: Component + Default>(
         |bencher| {
             bencher.iter_batched(
                 || {
-                    let mut world = setup::<Table>(entity_count);
+                    let mut world = setup::<T>(entity_count);
                     world.clear_trackers();
-                    let mut query = world.query::<&mut Table>();
-                    for mut table in query.iter_mut(&mut world) {
-                        black_box(&mut *table);
+                    let mut query = world.query::<&mut T>();
+                    for mut component in query.iter_mut(&mut world) {
+                        black_box(component.bench_modify());
```

Comment:
not sure about that `black_box`, probably doesn't hurt but as it is inside `bench_modify` now it's probably not useful here ?